ublox/src/ubx_packets/packets/sec_sig.rs

```diff
@@ -0,0 +1,150 @@
+//! SEC-SIG: Signal Security Status
+//!
+//! Provides real-time jamming and spoofing detection status.
+//! Critical for safety-critical autonomous systems.
+
+#[cfg(feature = "serde")]
+use super::SerializeUbxPacketFields;
+#[cfg(feature = "serde")]
+use crate::serde::ser::SerializeMap;
+
+#[allow(unused_imports, reason = "It is only unused in some feature sets")]
+use crate::FieldIter;
+use crate::{error::ParserError, UbxPacketMeta};
+use ublox_derive::{ubx_extend, ubx_packet_recv};
+
+/// Signal Security Status
+///
+/// Provides real-time jamming and spoofing detection status from
+/// the receiver's built-in interference detection capabilities.
+/// Essential for safety-critical GNSS applications.
+#[ubx_packet_recv]
+#[ubx(class = 0x27, id = 0x09, max_payload_len = 1028)] // 4 + jamNumCentFreqs*4 (max UBX payload length)
+struct SecSig {
+    /// Message version
+    version: u8,
+
+    /// Signal security flags
+    #[ubx(map_type = SecSigFlags)]
+    sig_sec_flags: u8,
+
+    /// Reserved
+    reserved0: u8,
+
+    /// The number of center frequencies we provide jamming information for
+    jam_num_cent_freqs: u8,
+
+    /// Jamming state of signals sharing a given center frequency (repeated jam_num_cent_freqs times)
+    #[ubx(map_type = SecSigJamStateCentFreqIter, may_fail,
+          from = SecSigJamStateCentFreqIter::new,
+          is_valid = SecSigJamStateCentFreqIter::is_valid)]
+    jam_state_cent_freqs: [u8; 0],
+}
+
+/// Jamming state for a given center frequency.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize))]
+pub struct SecSigJamStateCentFreq {
+    /// Center frequency in kHz (floored to nearest kHz)
+    pub cent_freq_khz: u32,
+    /// Flag indicates whether signals on the given center frequency are considered jammed
+    pub jammed: bool,
+}
+
+impl From<u32> for SecSigJamStateCentFreq {
+    fn from(value: u32) -> Self {
+        Self {
+            cent_freq_khz: value & 0x00ff_ffff,
+            jammed: (value & 0x0100_0000) != 0,
+        }
+    }
+}
+
+/// Iterator for SEC-SIG jam state entries.
+#[derive(Debug, Clone)]
+pub struct SecSigJamStateCentFreqIter<'a> {
+    data: &'a [u8],
+    offset: usize,
+}
+
+impl<'a> SecSigJamStateCentFreqIter<'a> {
+    fn new(data: &'a [u8]) -> Self {
+        Self { data, offset: 0 }
+    }
```

Comment:
Tiny thing, but giving the lifetime a slightly less generic name would be nice.

I just suggest `d` for `data`.

```suggestion
pub struct SecSigJamStateCentFreqIter<'d> {
    data: &'d [u8],
    offset: usize,
}

impl<'d> SecSigJamStateCentFreqIter<'d> {
    fn new(data: &'d [u8]) -> Self {
        Self { data, offset: 0 }
    }
```